js/src/builtin/streams/ReadableByteStreamController.cpp

```diff
@@ -761,15 +567,231 @@ enum BYOBRequestSlots {
   unwrappedBYOBRequest->setFixedSlot(BYOBRequestSlot_Controller,
                                      UndefinedValue());
 
-  // Step 3: Set controller.[[byobRequest]].[[view]] to undefined.
+  // // Step 3: Set controller.[[byobRequest]].[[view]] to undefined.
   unwrappedBYOBRequest->setFixedSlot(BYOBRequestSlot_View, UndefinedValue());
 
-  // Step 4: Set controller.[[byobRequest]] to undefined.
-  unwrappedController->clearBYOBRequest();
+  // // Step 4: Set controller.[[byobRequest]] to undefined.
+  unwrappedController->clearByobRequest();
 
   return true;
 }
 
 // Streams spec, 3.13.25.
 //      ReadableByteStreamControllerShouldCallPull ( controller )
 // Unified with 3.10.3 above.
+
+// https://streams.spec.whatwg.org/#abstract-opdef-readablebytestreamcontrollergetbyobrequest
+[[nodiscard]] static bool ReadableByteStreamController_byobRequest(
+    JSContext* cx, unsigned argc, Value* vp) {
+  CallArgs args = CallArgsFromVp(argc, vp);
+  Rooted<ReadableByteStreamController*> unwrappedController(
+      cx, UnwrapAndTypeCheckThis<ReadableByteStreamController>(
+              cx, args, "get byobRequest"));
+  if (!unwrappedController) {
+    return false;
+  }
+
+  Rooted<ListObject*> unwrappedPendingPullIntos(
+      cx, unwrappedController->pendingPullIntos());
+
+  if (!unwrappedController->hasByobRequest() &&
```

Comment:
```suggestion
// Step 1. If controller.[[byobRequest]] is null and controller.[[pendingPullIntos]] is not empty
  if (!unwrappedController->hasByobRequest() &&
```

Reply:
I stopped using step numbering, as strictly speaking we have two streams spec versions here - the byob on the latest spec, and the old streams on an older spec with different step numbers. New streams work doesn't include step numbers, while the old does. Totally not a confusing state of affairs.